addons/account_peppol/models/res_partner.py

```diff
@@ -37,10 +37,15 @@ class ResPartner(models.Model):
         copy=False,
     ) # field to compute the label to show for partner endpoint
 
-    @api.depends('peppol_eas', 'peppol_endpoint')
+    @api.depends('peppol_eas', 'peppol_endpoint', 'ubl_cii_format')
     def _compute_account_peppol_is_endpoint_valid(self):
-        # Every change in peppol_eas or peppol_endpoint should set the validity back to False
-        self.account_peppol_is_endpoint_valid = False
+        for partner in self:
+            if partner.ubl_cii_format in (False, 'facturx', 'oioubl_201'):
+                partner.account_peppol_is_endpoint_valid = False
+                continue
+
+            partner.button_account_peppol_check_partner_endpoint()
```

Comment:
You can't call this method here because it's setting a value to fields outside of the scope of this compute (`account_peppol_validity_last_check`)